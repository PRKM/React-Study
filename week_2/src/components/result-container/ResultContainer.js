import React from 'react';
import PropTypes from 'prop-types';

import './ResultContainer.css';

export default class ResultContainer extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        const { searchData, filterText } = this.props;
        return (
            <ul>
                {
                    searchData.filter(s => s.match(new RegExp(filterText, 'gi')))
                        .map((s, i) => {
                            return (
                                <li key={i}>{s}</li>
                            )
                        })
                }
            </ul>
        )
    }
}

ResultContainer.propTypes = {
    filterText: PropTypes.string,
    searchData: PropTypes.array
};