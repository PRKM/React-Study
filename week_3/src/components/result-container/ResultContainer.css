th {
    text-align: left;
}

td, th {
    padding: 2px 4px;
}