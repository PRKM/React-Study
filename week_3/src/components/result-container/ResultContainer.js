import React from 'react';

import './ResultContainer.css';

export default class ResultContainer extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            data: []
        };
    }

    componentWillMount() {
        fetch('https://raw.githubusercontent.com/mledoze/countries/master/countries.json')
            .then(res => {
                res.json()
                    .then(data => {
                        data.forEach(o => o.country = o.name.official);
                        this.setState({data: data});
                    });
            });
    }

    render() {
        let data = Object.assign(this.state.data);
        const {country, capital, region, subregion} = this.props.searchText;

        data = data.filter(o => {
            return (
                o.name.official.match(new RegExp(country), 'gi') &&
                o.capital.toString().match(new RegExp(capital), 'gi') &&
                o.region.match(new RegExp(region), 'gi') &&
                o.subregion.match(new RegExp(subregion), 'gi')
            );
        });

        return (
            <table>
                <thead>
                    <tr>
                        <th>Country</th>
                        <th>Capital</th>
                        <th>Region</th>
                        <th>Subregion</th>
                        <th>Latitude</th>
                        <th>Longtitude</th>
                    </tr>
                </thead>
                <tbody>
                    {
                        data.map((o, i) => {
                            return (
                                <tr key={i}>
                                    <td>{o.country}</td>
                                    <td>{o.capital.toString()}</td>
                                    <td>{o.region}</td>
                                    <td>{o.subregion}</td>
                                    <td>{o.latlng[0]}</td>
                                    <td>{o.latlng[1]}</td>
                                </tr>
                            )
                        })
                    }
                </tbody>
            </table>
        )
    }
}