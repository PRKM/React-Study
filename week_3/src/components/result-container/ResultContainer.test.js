import React from 'react';
import {shallow} from 'enzyme';

import ResultContainer from './ResultContainer';

import 'isomorphic-fetch';

describe('<ResultContainer/>', () => {
    let searchText;

    beforeAll(() => {
        searchText = {
            country: '',
            capital: '',
            region: '',
            subregion: ''
        };
    })

    test('Init state', () => {
        searchText = {
            country: '',
            capital: '',
            region: '',
            subregion: ''
        };
        let component = shallow(
            <ResultContainer searchText={searchText}/>,
        );
        
        expect(component.prop('searchText')).toBeDefined();
    });
})