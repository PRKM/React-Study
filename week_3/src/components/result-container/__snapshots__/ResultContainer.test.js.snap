// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultContainer/> Init state 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Country
      </th>
      <th>
        Capital
      </th>
      <th>
        Region
      </th>
      <th>
        Subregion
      </th>
      <th>
        Latitude
      </th>
      <th>
        Longtitude
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;
